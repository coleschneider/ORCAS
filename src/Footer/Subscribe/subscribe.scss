@import '../../scss/colors';

.input-feedback {
    color: $form-error;
}
 .footer__subscribe .subscribe-form {
  margin: 4rem 0 0 0;
  padding: 0;
  position: relative;
}

.footer__subscribe #mc-form {
  width: 100%;
}

.footer__subscribe #mc-form input[type="email"] {
  width: 100%;
  height: 6.8rem;
  font-size: 1.8rem;
  line-height: 3.2rem;
  padding: 1.8rem 140px 1.8rem 64px;
  background: url(../../icons/icon-mail.svg) 24px center no-repeat, rgba(255, 255, 255, 0.03);
  background-size: 24px 16px, auto;
  color: white;
  margin-bottom: 2rem;
  border: none;
  border-radius: 4px;
}

.footer__subscribe #mc-form input[type="email"]:hover,
.footer__subscribe #mc-form input[type="email"]:focus {
  box-shadow: none;
}

.footer__subscribe #mc-form input[type="email"]::-webkit-input-placeholder {

  /* WebKit, Blink, Edge */
  color: rgba(255, 255, 255, 0.3);
}

.footer__subscribe #mc-form input[type="email"]:-moz-placeholder {

  /* Mozilla Firefox 4 to 18 */
  color: rgba(255, 255, 255, 0.3);
  opacity: 1;
}

.footer__subscribe #mc-form input[type="email"]::-moz-placeholder {

  /* Mozilla Firefox 19+ */
  color: rgba(255, 255, 255, 0.3);
  opacity: 1;
}

.footer__subscribe #mc-form input[type="email"]:-ms-input-placeholder {

  /* Internet Explorer 10-11 */
  color: rgba(255, 255, 255, 0.3);
}

.footer__subscribe #mc-form input[type="email"]::-ms-input-placeholder {

  /* Microsoft Edge */
  color: rgba(255, 255, 255, 0.3);
}

.footer__subscribe #mc-form input[type="email"]::placeholder {

  /* Most modern browsers support this now. */
  color: rgba(255, 255, 255, 0.3);
}

.footer__subscribe #mc-form input[type="email"].placeholder {
  color: rgba(255, 255, 255, 0.3) !important;
}

.footer__subscribe #mc-form input[type=submit] {
  color: #ffffff;
  background: #80d3d2;
  border-color: #80d3d2;
  margin-right: 0;
  padding: 0 20px 0 24px;
  height: 6.8rem;
  line-height: 6.4rem;
  border-radius: 0 4px 4px 0;
  position: absolute;
  top: 0;
  right: 0;
}

.footer__subscribe #mc-form label {
  color: #ffffff;
  font-family: "Nunito Sans", sans-serif;
  font-size: 1.3rem;
  padding: 0 2rem;
}

.footer__subscribe #mc-form label i {
  padding-right: 6px;
}
