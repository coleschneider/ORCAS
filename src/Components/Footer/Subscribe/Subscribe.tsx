import * as React from 'react';
import { Formik } from 'formik';
import { ToastContainer, toast } from 'react-toastify';
import './subscribe.scss';
import { validateSubscribe } from 'utils/validate';
import instance from 'utils/apiService';

interface FormFields {
  email_address?: string;
}
interface NewsletterState {
  isFetching: boolean;
  error: null | Error;
}
class NewsletterForm extends React.Component<{}, NewsletterState> {
  constructor(props) {
    super(props);
    this.state = {
      error: null,
      isFetching: false,
    };
  }
  notifySuccess = () => toast.success(
    `Successfully subscribed to newsletter`,
    {
      autoClose: 5000,
      className: 'notification-inactive',
      position: 'bottom-right',
    },
  )
  notifyError = () => toast.warn(
    `There was an error attempting to subscribe to newsletter`,
    {
      autoClose: 5000,
      className: 'notification-inactive',
      position: 'bottom-right',
    },
  )
  subscribeUser = async (values: FormFields) => {
    this.setState({
      isFetching: true,
    });
    return instance
      .post('/', values)
      .then(() => {
        this.setState({
          isFetching: false,
        });
        this.notifySuccess()
      })
      .catch(error => {
        this.notifyError()
        this.setState({
          error,
          isFetching: false,
        });
      });
  };
  render() {
    return (
      <div className="subscribe-form">
        <Formik initialValues={{ email_address: '' }} onSubmit={this.subscribeUser} validate={validateSubscribe}>
          {({ values, touched, handleSubmit, errors, isSubmitting, handleChange, handleBlur, handleReset, dirty }) => {
            return (
              <form onSubmit={handleSubmit} id="mc-form" className="group">
                <input
                  type="email"
                  name="email_address"
                  onChange={handleChange}
                  onBlur={handleBlur}
                  value={values.email_address}
                  autoComplete="off"
                  className="email_address"
                  id="mc-email"
                  placeholder="Email Address"
                  className={errors.email_address && touched.email_address ? 'email text-input error' : 'email'}
                />
                {errors.email_address && touched.email_address && (
                  <div className="input-feedback">{errors.email_address}</div>
                )}

                <button type="submit" disabled={isSubmitting}>
                  Submit
                </button>
                <label className="subscribe-message" />
              </form>
            );
          }}
        </Formik>
        <ToastContainer />
      </div>
    );
  }
}

export default NewsletterForm;
