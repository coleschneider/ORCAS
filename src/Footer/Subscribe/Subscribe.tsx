import * as React from 'react';
import { withFormik, FormikActions } from 'formik';

import './subscribe.scss';

interface FormFields {
  email?: string;
}

const MyForm = props => {
  const { values, touched, errors, isSubmitting, handleChange, handleBlur, handleSubmit, handleReset, dirty } = props;

  return (
    <div className="subscribe-form">
      <form onSubmit={handleSubmit} id="mc-form" className="group">
        <input
          autoComplete="off"
          type="email"
          name="email"
          className="email"
          id="mc-email"
          placeholder="Email Address"
          value={values.email}
          onChange={handleChange}
          onBlur={handleBlur}
          className={errors.email && touched.email ? 'email text-input error' : 'email'}
        />
        {errors.email && touched.email && <div className="input-feedback">{errors.email}</div>}

        <input type="submit" name="subscribe" disabled={dirty || isSubmitting} value="Sign Up" />
        <label className="subscribe-message" />
      </form>
    </div>
  );
};

const MyEnhancedForm = withFormik({
  mapPropsToValues: () => ({ email: '' }),
  validate: values => {
    const errors: FormFields = {};
    if (!values.email) {
      errors.email = 'Required';
    } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
      errors.email = 'Invalid email address';
    }
    return errors;
  },

  handleSubmit: (values, { setSubmitting }) => {
    setTimeout(() => {
      alert(JSON.stringify(values, null, 2));
      setSubmitting(false);
    }, 1000);
  },

  displayName: 'SubscribeForm', // helps with React DevTools
})(MyForm);

export default MyEnhancedForm;
