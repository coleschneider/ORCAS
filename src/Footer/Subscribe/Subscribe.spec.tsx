import * as React from 'react';
import Subscribe from './Subscribe';

describe('Subscribe', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Subscribe />);
    expect(wrapper).toMatchSnapshot();
  });
});
