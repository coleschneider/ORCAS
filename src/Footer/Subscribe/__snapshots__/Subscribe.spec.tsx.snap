// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscribe should match the snapshot 1`] = `
<Formik
  displayName="SubscribeForm"
  enableReinitialize={false}
  handleSubmit={[Function]}
  initialValues={
    Object {
      "email": "",
    }
  }
  isInitialValid={false}
  onSubmit={[Function]}
  render={[Function]}
  validate={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
/>
`;
